### ISSUE TITLE
In popover if you shift tab past the trigger the popover stays open

### ISSUE THREAD

**Issue by jcfizzler** [chakra-ui/chakra-ui#6920]
### Description

In a popover if you shift tab past the trigger the popover stays open. Contrast this with tabbing through the popover, when you get to the last focusable element the popover closes. The user can no longer close the popover using a keyboard once they've left it using shift tab. They need to tab back to it and close using a close button or ESC once they've gone "too far". 

### Link to Reproduction

https://chakra-ui.com/docs/components/popover/usage

### Steps to reproduce

1. Go the popover example page.
2. Open the popover.
3. SHIFT + Tab back to the trigger. Note that due to issue https://github.com/chakra-ui/chakra-ui/issues/6821 you are sometimes  brought immediately back to the popover body again. But you can usually keep trying SHIFT tab to eventually get out. 
4. Notice the popover stays open though you have left it. And you cannot close it with a keyboard anymore without tabbing back into the popover.

### Chakra UI Version

v2.3.6

### Operating System

- [ ] macOS
- [ ] Windows
- [ ] Linux
